File: 프로그래머스연습문제/소수만들기.cpp
#include <vector>
#include <iostream>
#include <cmath>
using namespace std;

bool prime_number(int number){
    
    for(int i = 2; i <= sqrt(number); i++){
        if(number%i==0)return false;
    }
    return true;
}


int solution(vector<int> nums) {
    int answer = 0;
    int number = 0;
    
    for(int i = 0; i < nums.size()-2; i++){
        for(int j = i + 1; j < nums.size()-1; j++){
            for(int k = j + 1; k < nums.size(); k++){
                number = nums[i] + nums [j] + nums[k];
                cout << number <<endl;
                if(prime_number(number))answer++;

            }
        }
    }
    

    return answer;
}

File: 프로그래머스_python/연습문제/Level1/소수 만들기.py
from itertools import combinations
import math

def is_prime(number):
    
    if number == 1:
        return False
    
    flag = True
    for i in range(2,int(math.sqrt(number)) + 1):
        if number % i == 0:
            flag = False
            break
            
    return flag
    
def solution(nums):
    answer = 0

    array = list(combinations(nums,3))
    array_sum = []
    
    for i in range(len(array)):
        array_sum.append(array[i][0] + array[i][1] + array[i][2])
    
    
    
    
    for i in array_sum:
        if is_prime(i) == True:
            
            answer += 1
    
    
    

    return answer
